index.html

```diff
@@ -3749,17 +3781,8 @@ <h3>Data Validation</h3>
 											<ol>
 												<li id="rel-cover-first">
 													<p>if <var>cover</var> is <code>false</code>, set <var>cover</var>
-														to <code>true</code> and check:</p>
-													<ol>
-														<li id="rel-cover-name">
-															<p>if <var>resource["name"]</var> is not set or is an empty
-															string, <a>validation error</a>.</p>
-														</li>
-														<li id="rel-cover-desc">
-															<p>if <var>resource["description"]</var> is not set or is an
-															empty string, <a>validation error</a>.</p>
-														</li>
-													</ol>
+														to <code>true</code>. If <var>resource["name"]</var> is not set,
+															<a>validation error</a>.</p>
```

Comment:
The problem here is that the new text talks about the <var>name</var> requirement only for images:

> If the cover is an image (whether embedded in an HTML resource or not), it is strongly advised to follow Success Criterion 1.1.1   [  WCAG  ] for the provision of alternative text and extended descriptions. For image formats that do not provide the ability to embed this information, the  name  and  description  properties of  LinkedResource  can be used to provide alternative text and extended descriptions, respectively. In these cases, the  name  property  SHOULD  always be set — the property can be left empty for decorative images. 

while this check looks for <var>name</var> in all cases. Are we sure that any cover is an image? If so, the sentence above should be reformulated. If not, the check should be more complex, because it should then require a check on whether the URL is an image. And that is not obvious:

- either we require that the media type is set to an image (which may mean to list some media types that we consider as an image; can be done if we are careful to mention the fact that the media types can be extended in future) or
- the UA agent has to look into the file and find some ways of checking whether it is an image.

I am not sure which way we want to go. Checking the presence of <var>name</var> is a good thing, but making a complete check seems be fairly complex...

Reply:
Ya, my mind is toast right now. I meant to add an image requirement. Honest! :)

My thought is we limit ourselves to `encodingFormat`:

> If `encodingFormat` is set and specifies an image media type (`image/*`), and resource["name"] is not set, validation error.

There are various ways to work around this rule if you really want to, so I'm not sure it's necessary to dig in to file contents. The best enforcement is failing an accessibility standard, like WCAG, for those who have to take this requirement seriously, but that happens in a separate evaluation.

Reply:
Yeah, that sounds like right. It is not bulletproof, but it catches most of the possible situations and it takes care of the honest mistakes...